### PR TITLE
Master mrp back 2 basics marm

### PR DESCRIPTION
**Pull request by Dotni**
This back to basics fix and revamp some small features from mrp.

task-id: 3201527

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp_subcontracting/views/subcontracting_portal_templates.xml [hunk 1]
```diff
@@ -54,15 +58,19 @@
...
+            <t t-set="title" >Productions</t>
         </t>
+        <t t-set="no_footer" t-value="true"/>
+        <t t-call="mrp_subcontracting.subcontracting"/>
+    </t>
     </template>
 
     <template id="subcontracting" name="Subcontracting Portal View">
-        <iframe width="100%" height="100%" frameborder="0" t-attf-src="/my/productions/{{ picking_id }}/subcontracting_portal"/>
+        <iframe width="100%" height="700px" frameborder="0" t-attf-src="/my/productions/{{ picking_id }}/subcontracting_portal"/>
```
Comment (Dotni) [inline addons/mrp_subcontracting/views/subcontracting_portal_templates.xml:73]: Not sure that it's the correct solution, but setting the height to ``100%`` even in the parent container does not seem to work

Comment (AhmedAmirK) [reply addons/mrp_subcontracting/views/subcontracting_portal_templates.xml]: hmm I am not a 100% sure, but I think it should work if you add the `iframe` in a `div`

File: addons/mrp_subcontracting/views/subcontracting_portal_templates.xml [hunk 2]
```diff
@@ -54,15 +55,20 @@
...
+            <t t-set="breadcrumbs_url">/my/productions</t>
         </t>
+        <t t-set="no_footer" t-value="true"/>
+        <t t-call="mrp_subcontracting.subcontracting"/>
+    </t>
     </template>
 
     <template id="subcontracting" name="Subcontracting Portal View">
-        <iframe width="100%" height="100%" frameborder="0" t-attf-src="/my/productions/{{ picking_id }}/subcontracting_portal"/>
+        <iframe width="100%" height="700px" frameborder="0" t-attf-src="/my/productions/{{ picking.id }}/subcontracting_portal"/>
```
Comment (AhmedAmirK) [inline addons/mrp_subcontracting/views/subcontracting_portal_templates.xml:71]: So like something this should work
```suggestion
        <div>
            <iframe width="100%" height="100%" frameborder="0" t-attf-src="/my/productions/{{ picking.id }}/subcontracting_portal"/>
        </div>
```

File: addons/mrp/wizard/mrp_consumption_warning_views.xml [hunk 3]
```diff
@@ -38,7 +38,6 @@
                             colspan="1" type="object" class="btn-primary"/>
                         <button name="action_confirm" string="Confirm" invisible="consumption == 'strict'" data-hotkey="q"
                             colspan="1" type="object" class="btn-primary"/>
-                        <button name="action_set_qty" string="Set Quantities &amp; Validate" colspan="1" type="object" class="btn-secondary"/>
```
Comment (yhu-odoo) [inline addons/mrp/wizard/mrp_consumption_warning_views.xml:41]: This also remove it in our backend, not sure if we really want it.

File: addons/mrp_subcontracting/views/subcontracting_portal_templates.xml [hunk 4]
```diff
@@ -13,8 +13,9 @@
 
     <template id="portal_my_home_menu_production" name="Portal layout : production menu entries" inherit_id="portal.portal_breadcrumbs" priority="25">
         <xpath expr="//ol[hasclass('o_portal_submenu')]" position="inside">
-            <li t-if="page_name == 'production' or production" t-attf-class="breadcrumb-item #{'active' if not pickings else ''}">
-                <t>Productions</t>
+            <li t-if="title" t-attf-class="breadcrumb-item #{'active' if breadcrumbs_url else ''}">
```
Comment (yhu-odoo) [inline addons/mrp_subcontracting/views/subcontracting_portal_templates.xml:16]: image
image
Like on the purchase portal,  I think it's better to have `Home/ Production / XXX` and make `Production` clickable to go back to the list. What do you think?

Comment (Dotni) [reply addons/mrp_subcontracting/views/subcontracting_portal_templates.xml]: It's indeed better, it was the planned behavior of my changes but I think I managed to revert some of them(without me noticing it) and it was not working anymore.
Should be better now. 👌

File: addons/mrp/views/mrp_production_views.xml [hunk 5]
```diff
@@ -28,6 +28,7 @@
                         <button name="button_plan" type="object" string="Plan"/>
                         <button name="do_unreserve" type="object" string="Unreserve"/>
                         <button name="action_cancel" type="object" string="Cancel"/>
+                        <button name="action_assign"  type="object" string="Check availability"/>
```
Comment (yhu-odoo) [inline addons/mrp/views/mrp_production_views.xml:31]: For me, `Cancel` is normally the last button and it's also like that on the spec too.

File: addons/mrp/models/mrp_workorder.py [hunk 6]
```diff
@@ -14,6 +14,7 @@
 class MrpWorkorder(models.Model):
```
Comment (yhu-odoo) [inline addons/mrp/models/mrp_workorder.py:14]: Clean the `fixup` in the commit message :)

File: addons/website_sale/static/src/js/website_sale_delivery.js [hunk 7]
```diff
@@ -39,9 +39,10 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
             } else {
                 carrierChecked[0].click();
             }
+            await this._getCurrentLocation();
         }
 
-        await this._getCurrentLocation();
+
```
Comment (Feyensv) [inline addons/website_sale/static/src/js/website_sale_delivery.js:45]: what's this doing here ? :thinking:

Comment (Dotni) [reply addons/website_sale/static/src/js/website_sale_delivery.js:45]: Oops I apparently did not stash it 🙄

File: addons/mrp/wizard/mrp_consumption_warning_views.xml [hunk 8]
```diff
@@ -38,7 +38,7 @@
...
+                        <button name="action_set_qty" string="Set Quantities &amp; Validate" colspan="1" type="object" class="btn-secondary" invisible="1"/>
```
Comment (svs-odoo) [inline addons/mrp/wizard/mrp_consumption_warning_views.xml:41]: Should be invisible in portal only

**Comment by MiquelRForgeFlow** [comment]
This branch has conflicts

**Comment by svs-odoo** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**